librepo/downloader.c

```diff
@@ -1648,19 +1673,43 @@ check_transfer_statuses(LrDownload *dd, GError **err)
 
             if (!fatal_error &&
                 !complete_url_in_path &&
-                !target->target->baseurl &&
-                (dd->max_mirrors_to_try <= 0 ||
-                 num_of_tried_mirrors < dd->max_mirrors_to_try))
-            {
-                // Try another mirror
-                g_debug("%s: Ignore error - Try another mirror", __func__);
-                target->state = LR_DS_WAITING;
-                g_error_free(transfer_err);  // Ignore the error
-
-                // Truncate file - remove downloaded garbage (error html page etc.)
-                if (!truncate_transfer_file(target, err))
-                    return FALSE;
-            } else {
+                !target->target->baseurl)
+            {   
+              
+                // Another transfers are running or there are successful transfers
+                // and fewer failed transfers than tried parallel connections. It may be mirror is OK
+                // but accepts fewer parallel connections.
+                if (target->mirror->running_transfers > 0 ||
```

Comment:
downloader.c contains very long functions now. You prefer another extreme approach. A lot of very short (sometime one line) functions. Eg. has_running_transfers(mirror) will be contain only one line "return mirror->running_transfers > 0". Yes I know, short functions are more descriptive. OK, I can try change style of new code in downloader.c I will prepare new commit.